onadata/apps/api/viewsets/xform_submission_viewset.py

```diff
@@ -109,72 +49,24 @@ class XFormSubmissionViewSet(AuthenticateHeaderMixin,
-        username = self.kwargs.get('username')
-
-        if self.request.user.is_anonymous():
-            if username is None:
-                # raises a permission denied exception, forces authentication
-                self.permission_denied(self.request)
-            else:
-                user = get_object_or_404(User, username=username.lower())
-
-                profile, created = UserProfile.objects.get_or_create(user=user)
-
-                if profile.require_auth:
-                    # raises a permission denied exception,
-                    # forces authentication
-                    self.permission_denied(self.request)
-        elif not username:
-            # get the username from the user if not set
-            username = (request.user and request.user.username)
-
-        if request.method.upper() == 'HEAD':
-            return Response(status=status.HTTP_204_NO_CONTENT,
-                            headers=self.get_openrosa_headers(request),
-                            template_name=self.template_name)
-
-        is_json_request = is_json(request)
-
-        try:
-            create_fn = create_instance_from_json if is_json_request else\
-                create_instance_from_xml
-            error, instance = create_fn(username, request)
-        except AttributeError:
-            error = _(u'Incorrect format, see format details here, '
-                      u'https://api.ona.io/static/docs/submissions.html')
-
-        if error or not instance:
-            return self.error_response(error, is_json_request, request)
-
-        context = self.get_serializer_context()
-        serializer = SubmissionSerializer(instance, context=context)
-
-        return Response(serializer.data,
-                        headers=self.get_openrosa_headers(request),
-                        status=status.HTTP_201_CREATED,
-                        template_name=self.template_name)
-
-    def error_response(self, error, is_json_request, request):
-        if not error:
-            error_msg = _(u"Unable to create submission.")
-            status_code = status.HTTP_400_BAD_REQUEST
-        elif isinstance(error, basestring):
-            error_msg = error
-            status_code = status.HTTP_400_BAD_REQUEST
-        elif not is_json_request:
-            return error
-        else:
-            error_msg = xml_error_re.search(error.content).groups()[0]
-            status_code = error.status_code
-
-        return Response({'error': error_msg},
-                        headers=self.get_openrosa_headers(request),
-                        status=status_code)
+    def get_serializer_class(self):
+        if 'application/json' in self.request.content_type.lower():
+            self.request.accepted_renderer = JSONRenderer()
+            self.request.accepted_media_type = 'application/json'
+            return JSONSubmissionSerializer
+        if 'application/x-www-form-urlencoded' in self.request.content_type.lower():
```

Comment:
this is repeated on line 60, can refactor and express the structure better w/if else, e.g.

```python
content_type = self.request.content_type.ower()

if 'application/json' in content_type:
    ...
elif 'application/x-www-form-urlencoded' in content_type:
    ...
else
   return Submission Serializer
```